src/GraphBLAS-sharp.Backend/Common/ClArray.fs

```diff
@@ -162,16 +165,31 @@ module ClArray =
 
             bitmap
 
+    let getUniqueBitmapFirstOccurrence clContext =
+        getUniqueBitmapGeneral
+        <| <@ fun (gid: int) (_: int) (inputArray: ClArray<'a>) ->
+                  gid = 0
+                  || inputArray.[gid - 1] <> inputArray.[gid] @>
+        <| clContext
+
+    let getUniqueBitmapLastOccurrence clContext =
+        getUniqueBitmapGeneral
+        <| <@ fun (gid: int) (length: int) (inputArray: ClArray<'a>) ->
```

Comment:
Code duplication with [this](https://github.com/IgorErin/GraphBLAS-sharp/blob/96c0c0822b32ee0a737c292f5df65de213170472/src/GraphBLAS-sharp.Backend/Common/Scatter.fs#L6)